keras/backend/mxnet_backend.py

```diff
@@ -2555,7 +2557,78 @@ def rnn(step_function, inputs, initial_states,
         ValueError: if `mask` is provided (not `None`) but states is not provided
             (`len(states)` == 0).
     """
-    raise NotImplementedError('MXNet Backend: RNNs are not supported yet.')
+    if not unroll:
+        raise NotImplementedError('MXNet Backend: unroll=False '
+                                  'is not supported yet in RNN.')
+    dtype = inputs.dtype
+    dshape = inputs.shape
+    if len(dshape) < 3:
+        raise ValueError('Input tensor should be at least 3-D')
+    if not dshape[1]:
+        raise ValueError('Unrolling requires a fixed number of timesteps.')
+
+    # Split the inputs across time dimension and generate the list of inputs
+    # with shape `(samples, ...)` (no time dimension)
+    inputs = list(mx.sym.split(inputs.symbol, axis=1,
+                               squeeze_axis=1, num_outputs=dshape[1]))
+
+    # Reverse the input sequence
+    if go_backwards:
+        inputs.reverse()
+
+    # Assume learning phase is a placeholder tensor.(F = test, T = train)
+    global uses_learning_phase
```

Comment:
Some Keras layers (e.g. Dropout, BatchNormalization) behave differently at training time and testing time. uses_learning_phase is True if the layer has a different behavior in training mode and test mode, False otherwise.